arch/lkl/kernel/time.c

```diff
@@ -121,7 +119,7 @@ void __init time_init(void)
 	if (ret)
 		pr_err("lkl: unable to register clocksource\n");
 
-	clockevents_config_and_register(&clockevent, HZ, 0, 0xffffffff);
+	clockevents_config_and_register(&clockevent, NSEC_PER_SEC, 1, ULONG_MAX);
```

Comment:
Yes. What seems to happen is that actual delta is say 1032ns but clc is set to 0. The host timer will be set to 1ns so the timer irq will be shot very fast and the new delta will be say 980 but the clc will be again 0 which again will short a very fast timer irq, and so on.